skimage/feature/peak.py

```diff
@@ -41,6 +42,8 @@ def peak_local_max(image, min_distance=1, threshold_abs=None,
     num_peaks : int, optional
         Maximum number of peaks. When the number of peaks exceeds `num_peaks`,
         return `num_peaks` peaks based on highest peak intensity.
+    num_peaks_per_label : int, optional
+        Maximum number of peaks for each label.
```

Comment:
This is an API breaking change (until we switch to Python3 only and make these keyword-only arguments). I suggest you put this new argument at the end so we can avoid the deprecation.